### PR TITLE
Update vm-common and arwen

### PR DESCRIPTION
**Pull request by sstanculeanu**
Last piece of integration of enable epochs handler, into arwen and vm-common.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,16 +3,16 @@ module github.com/ElrondNetwork/elrond-go
 go 1.15
 
 require (
-	github.com/ElrondNetwork/arwen-wasm-vm/v1_2 v1.2.41
-	github.com/ElrondNetwork/arwen-wasm-vm/v1_3 v1.3.41
-	github.com/ElrondNetwork/arwen-wasm-vm/v1_4 v1.4.58
+	github.com/ElrondNetwork/arwen-wasm-vm/v1_2 v1.2.42-0.20220725094651-084c4dd97fc9
```
Comment (iulianpascalau) [inline go.mod:6]: Proper releases when the feature is finished

File: integrationTests/testProcessorNodeWithTestWebServer.go [hunk 2]
```diff
@@ -137,13 +137,14 @@ func createFacadeComponents(tpn *TestProcessorNode) (nodeFacade.ApiResolver, nod
...
-		MapDNSAddresses:           make(map[string]struct{}),
-		Marshalizer:               TestMarshalizer,
-		Accounts:                  tpn.AccntState,
-		ShardCoordinator:          tpn.ShardCoordinator,
+		GasSchedule:             gasScheduleNotifier,
+		MapDNSAddresses:         make(map[string]struct{}),
+		Marshalizer:             TestMarshalizer,
+		Accounts:                tpn.AccntState,
+		ShardCoordinator:        tpn.ShardCoordinator,
 		EpochNotifier:             tpn.EpochNotifier,
```
Comment (iulianpascalau) [inline integrationTests/testProcessorNodeWithTestWebServer.go:145]: misalignment here and on L148?

File: integrationTests/vm/testInitializer.go [hunk 3]
```diff
@@ -635,21 +635,22 @@ func CreateVMAndBlockchainHookMeta(
...
 		MapDNSAddresses: map[string]struct{}{
 			string(dnsAddr): {},
 		},
-		Marshalizer:               testMarshalizer,
-		Accounts:                  accnts,
-		ShardCoordinator:          shardCoordinator,
+		Marshalizer:             testMarshalizer,
+		Accounts:                accnts,
+		ShardCoordinator:        shardCoordinator,
 		EpochNotifier:             globalEpochNotifier,
```
Comment (iulianpascalau) [inline integrationTests/vm/testInitializer.go:647]: misalignment here and on L650

File: process/factory/shard/vmContainerFactory.go [hunk 4]
```diff
@@ -63,6 +63,9 @@ func NewVMContainerFactory(args ArgVMContainerFactory) (*vmContainerFactory, err
 	if check.IfNil(args.EpochNotifier) {
 		return nil, process.ErrNilEpochNotifier
 	}
+	if check.IfNil(args.EnableEpochsHandler) {
+		return nil, process.ErrNilEnableEpochsHandler
```
Comment (iulianpascalau) [inline process/factory/shard/vmContainerFactory.go:67]: unit test for this case?

File: process/smartContract/builtInFunctions/factory.go [hunk 5]
```diff
@@ -52,31 +45,25 @@ func CreateBuiltInFunctionsFactory(args ArgsCreateBuiltInFunctionContainer) (vmc
 	if check.IfNil(args.EpochNotifier) {
 		return nil, process.ErrNilEpochNotifier
 	}
+	if check.IfNil(args.EnableEpochsHandler) {
+		return nil, process.ErrNilEnableEpochsHandler
```
Comment (iulianpascalau) [inline process/smartContract/builtInFunctions/factory.go:49]: unit test?

Comment (sstanculeanu) [reply process/smartContract/builtInFunctions/factory.go:49]: added for all params

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**